code-review.js

```diff
@@ -1,60 +1,111 @@
+	 */
 	this.add = function(value, priority) {
-		if (this.store[priority] == undefined)
+		if (this.store[priority] == undefined) {
 			this.store[priority] = [];
+		}
 
 		this.store[priority].push(value);
 		this.count++;
 	};
 
-	// returns the oldest-added value with the highest priority
-	this.Pop = function() {
-		maxKey = Math.max(Object.keys(this.store));
+	/**
+	 * Returns the oldest-added value with the highest priority
+	 */
+	this.pop = function() {
+		if (this.count === 0) {
+			return false;
+		}
+		var maxKey = Math.max.apply(null, this.getAllPriorities());
+		var poppedValue = this.store[maxKey].shift();
 		this.count--;
-		return this.store[maxKey].shift();
+		this.removeEmptyPriority(maxKey);
+
+		return poppedValue
 	};
 
+	/**
+	 * Returns the number of elements in queue
+	 * @returns {Number} 
+	 */
 	this.length = function() {
 		return this.count;
-	}
-}
+	};
+};
 
-PriorityQueue.prototype.get_all_priorities = function() {
+/**
+ * Returns an array of numerical priorities
+ * @returns {Array}
+ */
+PriorityQueue.prototype.getAllPriorities = function() {
 	return Object.keys(this.store);
-}
+};
 
-// iterates through all the queue elements in priority-then-FIFO order
-PriorityQueue.prototype.forEach = function(callback) {
-	var keys = Object.keys(this.store).sort();
+/**
+ * Iterates through all the queue elements in priority-then-FIFO order
+ * 
+ * @param {function} callback - Callback function to be used on every element.
+ */
+PriorityQueue.prototype.priorityForEach = function(callback) {
+	for (var priority in this.store){
+		for (var i = 0; i < this.store[priority].length; i++) {
+			callback(this.store[priority][i]);
+		}
+	}
+};
 
-	for (var a = keys.length; a > 0; a--) {
-		for (var b = 0; b < this.store[a].length; b++)
-			callback(this.store[a][b]);
+/**
+ * Tests if a priority's array is empty and deletes the priority if not.
+ * 
+ * @param {Number} priority - Priority to be tested
+ */
+PriorityQueue.prototype.removeEmptyPriority = function(priority) {
+	if (this.store[priority].length === 0){
+		delete this.store[priority];
 	}
-}
+};
 
+/**
+ * Changes the Priority to a different key value.
+ *
+ * @param {*} value - The Item we wish to change
+ * @param {Number} newPriority - Positive integer 
+ */
 PriorityQueue.prototype.changePriority = function(value, newPriority) {
 	var foundItem = false;
 
-	this.store.forEach(function(bucket) {
-		bucket.forEach(function(item, index) {
+	for (var priority in this.store){
+		this.store[priority].forEach(function(item, index) {
 			if (item === value) {
-				bucket.splice(index, 1);  // remove the item
+				this.store[priority].splice(index, 1);  // remove the item
 				this.add(value, newPriority);
+				this.count--;
 				foundItem = true;
+				this.removeEmptyPriority(priority);
 				return false;  // early exit from forEach
 			}
-		});
-		if (foundItem) return false;
-	});
-}
\ No newline at end of file
+		}, this);
+		if (foundItem) {
+			return true;

```

Comment:
Return `true` on successfully finding item and return `false` if not. Before `false` is always returned.